dataland-frontend/tests/component/components/pages/QualityAssurance.cy.ts

```diff
@@ -0,0 +1,100 @@
+import { DataMetaInformation, DataTypeEnum, PathwaysToParisData, SmeData } from "@clients/backend";
+import { QaStatus } from "@clients/qaservice";
+import { assertDefined } from "@/utils/TypeScriptUtils";
+
+describe("Component tests for the Quality Assurance page", () => {
+  const keycloakMock = minimalKeycloakMock({
+    roles: ["ROLE_USER", "ROLE_UPLOADER", "ROLE_REVIEWER"],
+  });
+
+  /**
+   * Mock a data meta information object for a dataset specified by its fixture data, data Id and data type
+   * @param fixtureData The fixture data corresponding to review
+   * @param dataId The Id corresponding to the dataset to review
+   * @param dataType The type of the dataset to review
+   * @returns the data meta information object created
+   */
+  function mockQaDataObjectForDatasetWithId(
+    fixtureData: FixtureData<PathwaysToParisData | SmeData>,
+    dataId: string,
+    dataType: DataTypeEnum
+  ): DataMetaInformation {
+    return {
+      dataId: dataId,
+      companyId: "testCompanyId",
+      dataType: dataType,
+      uploadTime: 1672531200 /* 01.01.2023 */,
+      reportingPeriod: fixtureData.reportingPeriod,
+      currentlyActive: false,
+      qaStatus: QaStatus.Pending,
+    } as DataMetaInformation;
+  }
+
+  /**
+   * Create a QA data object from the fixture data and the data id, configure the intercepts, mount the component and
+   * check that clicking leads to the expected request
+   * @param fixtureData The fixture data corresponding to review
+   * @param dataId The Id corresponding to the dataset to review
+   * @param dataType The type of the dataset to review
+   */
+  function createQaDataObjectAndPerformRequestCheck(
+    fixtureData: FixtureData<PathwaysToParisData | SmeData>,
+    dataId: string,
+    dataType: DataTypeEnum
+  ): void {
+    const qaDataObject = {
+      dataId: dataId,
+      metaInformation: mockQaDataObjectForDatasetWithId(fixtureData, dataId, dataType),
+      companyInformation: fixtureData.companyInformation,
+    };
+    cy.intercept("**/qa/datasets", [dataId]);
+    cy.intercept("**/api/metadata/*", qaDataObject.metaInformation);
+    cy.intercept("**/api/companies/*", {
+      companyId: qaDataObject.metaInformation.companyId,
+      companyInformation: qaDataObject.companyInformation,
+      dataRegisteredByDataland: [qaDataObject.metaInformation],
+    });
+    let dataTypeString;
+    if (dataType === DataTypeEnum.P2p) {
+      dataTypeString = "p2p";
+    } else if (dataType === DataTypeEnum.Sme) {
+      dataTypeString = "sme";
+    }
+    cy.intercept(`**/api/data/${assertDefined(dataTypeString)}/*`, {
+      companyId: qaDataObject.metaInformation.companyId,
+      reportingPeriod: fixtureData.reportingPeriod,
+      data: fixtureData.t,
+    }).as("fetchData");
+    cy.mountWithPlugins<typeof QualityAssurance>(QualityAssurance, {
+      keycloak: keycloakMock,
+    }).then((mounted) => {
+      void mounted.wrapper.setData({
```

Comment:
The data you set in this setData are responsible for what is shown. Don't set them. I bet it works without this setData block.

Reply:
Indeed, not necessary. Checked and changed it